File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let top = 0;
    let bottom = 0;
    let prev = -(nums[0] + 1);

    while (top < nums.length) {
        if (nums[top] != prev) {
            nums[bottom] = nums[top];
            bottom++;
        }
        prev = nums[top];
        top++;
    }
    return bottom;
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize) {
   int* top = nums;
   int* bottom = nums;
   int prev = -((*top) + 1);
   while (top - nums < numsSize) {
       if (*top != prev) {
           *bottom = *top;
           bottom++;
       }
       prev = *top;
       top++;
   }
   return bottom-nums;
}